src/Symfony/Component/Console/Helper/QuestionHelper.php

```diff
@@ -461,6 +461,11 @@ private function validateAttempts(callable $interviewer, OutputInterface $output
     {
         $error = null;
         $attempts = $question->getMaxAttempts();
+
+        if (null === $attempts && !$this->isTty()) {
```

Comment:
What I don't understand is, why run stty at all if there is no tty? Running it will cause `stty: stdin isn't a terminal` error messages

Reply:
I've silenced those messages in this patch, but indeed you are right. Why do we still attempt to read here? Because despite the message, it apparently still works and returns content in stdin.

Reply:
Ah, right! I somehow managed to forget about that. Then this change indeed fixes the infinite loop I ran into.